cli/compose/convert/compose.go

```diff
@@ -97,6 +97,10 @@ func Networks(namespace Namespace, networks networkMap, servicesNetworks map[str
 func Secrets(namespace Namespace, secrets map[string]composetypes.SecretConfig) ([]swarm.SecretSpec, error) {
 	result := []swarm.SecretSpec{}
 	for name, secret := range secrets {
+		if secret.Name != "" {
+			name = secret.Name
+		}
```

Comment:
This is not used if `secret.External.External` evaluates to true, so I think this block should move after the next `if secret.External.External` block. 

Also, this name should not be scoped using `namespace.Scope(name)` (which is done below). So there needs to be an `else` for doing the `namespace.Scope()`


Same for configs below (I wish these two types shared more code...)